challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map { |x| x.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select { |x| x.downcase == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.select { |x| x.downcase != "dog" }
```

Comment:
This is legit, but there's also a [`reject`](http://ruby-doc.org/core-2.4.2/Enumerable.html#method-i-reject) method.